immich-app/immich
# feat(server): asset entity audit

continues from #3592 (rebased)

## Features:
- [x] Deletes automatically populated by TypeORM hooks (thanks @alextran1502 )
- [x] Get deletes after a certain date (with check whether a full sync is required)
- [x] Asset API endpoint returns changed assets via `updatedAfter` query param
- [x] New nightly job to clean audit log table entries older than 100 days

## Tested Scenarios:
- Favorite an asset in the web, view updated date is updated
- Unfavorite an asset in the web, view updated date is updated
- Archive an asset in the web, view updated date is updated
- Unarchive an asset in the web, view updated date is updated
- Delete an asset in the web, view audit log is created
- Delete an album in the web, view audit log is created
- Query deletes with old date (> 100 years) and see `needsFullSync` is true
- Query assets by updatedAt and view recently updated assets are returned
- Run refresh metadata on a single asset and verify updated date is changed
- Upload a new asset and verify asset is returned in recent updates query
